src/overlays/kaleido_scope/ovl_kaleido_scope/z_kaleido_mask.c

```diff
@@ -1,59 +1,808 @@
+            if (!CHECK_MOON_MASK_TAKEN(i)) {
+                if ((pauseCtx->unk_200 == 0) && (pauseCtx->pageIndex == PAUSE_MASK) &&
+                    (pauseCtx->cursorSpecialPos == 0) &&
+                    gMaskPlayerFormSlotRestrictions[(void)0, gSaveContext.save.playerForm][i]) {
+                    if ((sMaskEquipState == EQUIP_STATE_MAGIC_ARROW_HOVER_OVER_BOW_SLOT) && (i == SLOT_ARROW_ICE)) {
+                        // Possible bug:
+                        // Supposed to be `SLOT_BOW`, unchanged from OoT, instead increase size of ice arrow icon
+                        gDPSetPrimColor(POLY_OPA_DISP++, 0, 0,
+                                        sMaskMagicArrowEffectsR[pauseCtx->equipTargetItem - 0xB5],
+                                        sMaskMagicArrowEffectsG[pauseCtx->equipTargetItem - 0xB5],
+                                        sMaskMagicArrowEffectsB[pauseCtx->equipTargetItem - 0xB5], pauseCtx->alpha);
+
+                        pauseCtx->maskVtx[j + 0].v.ob[0] = pauseCtx->maskVtx[j + 2].v.ob[0] =
+                            pauseCtx->maskVtx[j + 0].v.ob[0] - 2;
+                        pauseCtx->maskVtx[j + 1].v.ob[0] = pauseCtx->maskVtx[j + 3].v.ob[0] =
+                            pauseCtx->maskVtx[j + 0].v.ob[0] + 32;
+                        pauseCtx->maskVtx[j + 0].v.ob[1] = pauseCtx->maskVtx[j + 1].v.ob[1] =
+                            pauseCtx->maskVtx[j + 0].v.ob[1] + 2;
+                        pauseCtx->maskVtx[j + 2].v.ob[1] = pauseCtx->maskVtx[j + 3].v.ob[1] =
+                            pauseCtx->maskVtx[j + 0].v.ob[1] - 32;
+
+                    } else if (i == pauseCtx->cursorSlot[PAUSE_MASK]) {
+                        // Increase the size of the selected item
+                        pauseCtx->maskVtx[j + 0].v.ob[0] = pauseCtx->maskVtx[j + 2].v.ob[0] =
+                            pauseCtx->maskVtx[j + 0].v.ob[0] - 2;
+                        pauseCtx->maskVtx[j + 1].v.ob[0] = pauseCtx->maskVtx[j + 3].v.ob[0] =
+                            pauseCtx->maskVtx[j + 0].v.ob[0] + 32;
+                        pauseCtx->maskVtx[j + 0].v.ob[1] = pauseCtx->maskVtx[j + 1].v.ob[1] =
+                            pauseCtx->maskVtx[j + 0].v.ob[1] + 2;
+                        pauseCtx->maskVtx[j + 2].v.ob[1] = pauseCtx->maskVtx[j + 3].v.ob[1] =
+                            pauseCtx->maskVtx[j + 0].v.ob[1] - 32;
+                    }
+                }
+
+                gSPVertex(POLY_OPA_DISP++, &pauseCtx->maskVtx[j + 0], 4, 0);
+                KaleidoScope_DrawQuadTextureRGBA32(
+                    play->state.gfxCtx, gItemIcons[((void)0, gSaveContext.save.inventory.items[i + NUM_ITEM_SLOTS])],
+                    32, 32, 0);
+            }
+        }
+    }
+
+    CLOSE_DISPS(play->state.gfxCtx);
+}
+
+u8 sMaskPlayerFormItems[] = {
+    ITEM_MASK_FIERCE_DEITY, ITEM_MASK_GORON, ITEM_MASK_ZORA, ITEM_MASK_DEKU, ITEM_NONE,
 };
+void KaleidoScope_UpdateMaskCursor(PlayState* play) {
```

Comment:
```suggestion
};

void KaleidoScope_UpdateMaskCursor(PlayState* play) {
```
as before, although I suspect this is in-function static?